src/afio.rs

```diff
@@ -10,13 +10,16 @@ use crate::gpio::{
 };
 
 pub trait AfioExt {
-    fn constrain(self, apb2: &mut APB2) -> Parts;
+    fn constrain(self) -> Parts;
 }
 
 impl AfioExt for AFIO {
-    fn constrain(self, apb2: &mut APB2) -> Parts {
-        AFIO::enable(apb2);
-        AFIO::reset(apb2);
+    fn constrain(self) -> Parts {
+        unsafe {
+            let rcc = &(*RCC::ptr());
+            AFIO::enable(rcc);
+            AFIO::reset(rcc);
+        }
```

Comment:
As far as I can tell, without doing this, any use of AFIO will be pointless, but in some of the examples, we no longer constrain AFIO.

Note to self: Is this the case?

Reply:
Not sure what you mean, but `&mut afio.mapr` stayed where it was. I've not deleted it.

Reply:
I was (wrongly) under the impression that AFIO was used for some GPIO stuff, but that seems to not be the case. Disregard this comment